bin/publish_pod

```diff
@@ -10,6 +10,9 @@ if [ -n "$BUILDKITE_TAG" ] && [ "$BUILDKITE_TAG" != "$POD_VERSION" ]; then
 	exit 1
 fi
 
+echo "⚠️ Remove the patch-cocoapods step once this issue is fixed: https://github.com/CocoaPods/CocoaPods/issues/12033"
+patch-cocoapods
```

Comment:
I wonder if we shouldn't instead let the caller projects call `patch-cocoapods` before they call `validate_pod` and `publish_pod` in their own `.buildkite/commands/*.sh` scripts.

The reason I suggest separating those is because the patch depends on the version of CocoaPods being installed. If/when CocoaPods releases a new version with this issue fixed, some projects will adopt the new version quickly while others might take more time.

At least by letting the caller script decide, it will be easier once the new CP version arrives to just update CP in the project's `Gemfile.lock` then stop calling `patch-cocoapods` on the buildkite scripts in the same PR updating CP, rather than having to release a new version of a8c-ci-toolkit that will remove that patch's call… but then only work with projects that updated CocoaPods.

If we strongly couple the version of a8c-ci-toolkit with the version of CocoaPods used, if some projects want to update CocoaPods but are not ready to update a8c-ci-toolkit (eg say there was a major version of the toolkit since for unrelated changes, and the project isn't ready to adopt those yet) they'd be stuck.

Reply:
The main reason for putting this script here is: `install_gems` overwrites the local gems, so `patch-cocoapods` needs to run straight after `install_gems`. Also, it'd be nice to avoid copying scripts among libraries repos.

But I agree with you it's not ideally that this script may break pipelines.

Initially I tried to add an optional plugin configuration to the plugin, `patch-cocoapods: true`, but I can't get the environment variable `BUILDKITE_PLUGIN_A8C_CI_TOOLKIT_PATCH_COCOAPODS` to work: it doesn't exist even though [the new option is set in pipeline](https://github.com/wordpress-mobile/WordPressKit-iOS/commit/0a960998625c3c2c4b6488c24aab0564d0ea3a73).

What do you think adding a new option to the existing shell script, like `validate_podspec --patch_cocoapods`?

Reply:
> The main reason for putting this script here is: install_gems overwrites the local gems, so patch-cocoapods needs to run straight after install_gems. Also, it'd be nice to avoid copying scripts among libraries repos.

Ah. Damn.

> Initially I tried to add an optional plugin configuration to the plugin, `patch-cocoapods: true`, but I can't get the environment variable `BUILDKITE_PLUGIN_A8C_CI_TOOLKIT_PATCH_COCOAPODS` to work: it doesn't exist even though [the new option is set in pipeline](https://github.com/wordpress-mobile/WordPressKit-iOS/commit/0a960998625c3c2c4b6488c24aab0564d0ea3a73).

Just checking: did you declare the additional plugin config parameter in `plugin.yml` [here](https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/blob/trunk/plugin.yml#L12-L15) (then pointed to the commit that added it in your WPKiOS pipeline) when you attempted that?

iinm, the plugin has `additionalProperties: false` in its `plugin.yml` config, so all additional properties have to be declared explicitly in `plugin.yml` for them to be exposed as env var and then accessible within the plugin.

That being said…

> What do you think adding a new option to the existing shell script, like validate_podspec --patch_cocoapods?

I actually think that's probably a better approach 👍  so that the option is tied to the commands that need it instead of to the whole plugin (especially wouldn't make sense to expose that option to app / non-pod repos, which use the a8c-ci-toolkit plugin… but never call `validate_pod` nor `publish_pod` in their context)

Reply:
> did you declare the additional plugin config parameter in plugin.yml

Yes, I did. I tried naming it as `patch-cocoapods`, `patch_cocoapods`. I tried declaring it as string, boolean. None worked...

----

I have implemented the new `--patch_cocoapods` argument in the new commits. And [this test job](https://buildkite.com/automattic/wordpresskit-ios/builds/1152#018c18f7-6f8d-46f8-8012-94234f930b27) passed.